### ISSUE TITLE
Background network processing

### ISSUE THREAD

**Issue by gedu** [Expensify/App#50140]
## Problem
When a user opens the chat app after a period of inactivity/activity/actions, the NetworkQueue is filled with a large queue of unsent requests. This queue causes significant processing delays, which result in app hangs and infinite loading of messages, preventing users from having a seamless messaging experience.

## Solution
To address this problem, consider implementing background task processing that allows the app to handle network requests even when not actively open. For mobile (Android/iOS), you could utilize:

For Android: Use WorkManager to manage background tasks reliably.
For iOS: Implement Background Fetch to periodically process network requests.
For both: Consider using React Native Background Fetch or https://reactnative.dev/docs/headless-js-android

### Additional Solutions for Web and Desktop
Web: Leverage Service Workers for handling background sync when internet connection is re-established.
Desktop: Use Electron’s background task handling capabilities (e.g., electron-ipc).

### Cases:
1. User is offline or in a poor network connection
2. User takes some actions and serializes network requests
3. User puts app in the background
4. User regains network connection, requests from the queue start executing.
5. User opens the app and has no remaining network requests (or maybe a few)

1. Open the app from a notification
2. Read it
3. Write a response
4. Switch apps

This background processing will prevent a message from not being sent due to moving quickly to the background.

**Comment by szymonrybczak** [comment]
Hey, I'm Szymon from Callstack and I can work on this 🫡

**Comment by szymonrybczak** [comment]
Hey, I started working on this issue and in first place I looked on what are possible libraries that we leverage to solve this problem. I've started with Native platforms. There are not many tools that are solving this problem well, and actually there are two that I found that are maintained: 
- https://github.com/transistorsoft/react-native-background-fetch 
    - no support for New Architecture with latest versions of React Native,
	- API may be outdated due to long-term development, containing legacy code and patterns
	- longer history of use
- https://docs.expo.dev/versions/latest/sdk/background-fetch/
    - supports New Architecture with latest React Native versions
    - uses Expo Modules under the hood which are already used in the project
    - modern & more stable API

I'd go with `expo-background-fetch` together with `expo-task-manager` which overall fits as more reliable and better option for this use case.

**Comment by szymonrybczak** [comment]
Hey, update from my side. I successfully implemented PoC of registering background task with `expo-task-manager` on iOS. Right now I'm focusing on doing the same on Android, there's one crash when app starts - I'm trying to investigate why is it happening.

**Comment by gedu** [comment]
Updated P/S

**Comment by szymonrybczak** [comment]
Hey, just to update this issue with current status - we'll be developing our own implementation of executing background task. Here's a separate issue with all necessary informations: https://github.com/Expensify/App/issues/52700

**Comment by luacmartins** [comment]
Both iOS and android PRs are merged. @gedu is there any more work in progress associated with this issue?

**Comment by mananjadhav** [comment]
@luacmartins @mountiny Can you assign a BZ for the payout here?

**Comment by mananjadhav** [comment]
@laurenreidexpensify Quick bump on the payment summary.

**Comment by laurenreidexpensify** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by laurenreidexpensify** [comment]
Payment Summary: 

- @shubham1206agra paid $250 Upwork (Contributor)
- @szymonrybczak does not require payment (Expert Contribtor)
- @mananjadhav reviewer $250 can request via ND (c+)

**Comment by laurenreidexpensify** [comment]
@mananjadhav sorry for delay - this was a weekly so not on my immediate radar. 
Please complete checklist before we can close thanks

**Comment by mananjadhav** [comment]
This was more of a non-customer issue. There's no specific regression test suite and can be tagged as feature request.

**Comment by laurenreidexpensify** [comment]
Closing as no further action required

**Comment by garrettmknight** [comment]
$250 approved for @mananjadhav
